proposals/0082-relax-transaction-constraints.md

```diff
@@ -0,0 +1,173 @@
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+Specifically, this proposal relaxes constraints that a transaction included in
+a block must:
+
+1. Have a fee-payer with enough funds to pay fees
+2. Have a valid nonce account, if specified
+3. Have program accounts that:
+   1. exist
+   2. are executable
+4. Have writable accounts that are:
+   1. not executable, unless owned by
+   `BPFLoaderUpgradeab1e11111111111111111111111`
+   2. if owned by `BPFLoaderUpgradeab1e11111111111111111111111`,
+   then `BPFLoaderUpgradeab1e11111111111111111111111` must be included
+   3. if owned by `Stake11111111111111111111111111111111111111`, then the
+   current slot must not be within the epoch stakes reward distribution period
+5. Have executable `BPFLoaderUpgradeab1e11111111111111111111111` owned accounts
+with `UpgradeableLoaderState::Program` state, and derived program data account
+that exist
+6. Have a call chain depth of 5 or less
+7. Have no builtin loader ownership chains
+(pending `4UDcAfQ6EcA6bdcadkeHpkarkhZGJ7Bpq7wTAiRMjkoi`)
+8. Have total loaded data size which does not exceed
+requested_loaded_accounts_data_size_limit
+(pending `DdLwVYuvDz26JohmgSbA7mjpJFgX5zP2dkp8qsF2C33V`)
+
+The intent with relaxing these constraints is to minimize the amount of
+account-state which is required in order to validate a block. This gives more
+flexibility to when and how account-state is updated during both block
+production and validation.
+
+With these constraints removed, there is still one account-state constraint
+at the transaction level: address lookup table resolution.
+With this proposal, this constraint is intentionally not relaxed since it is
+necessary for validation of entry-level and block-level constraints.
+However, if/when those constraints are relaxed, this constraint should be
+relaxed as well.
+
+The relaxation of these constraints is only relaxed for transactions being
+included in a block. During block validation, if any of these constraints
+are broken, the entire block is marked invalid; with this proposal, the
+block is not marked invalid, but the transaction will not be executed.
+These constraints must still be satisfied in order for the transaction to be
+executed, and have an effect on state. However, there are some new considerations
+which must be taken into account, specifically as it relates to fees and block
+limits.
+
+### Fee-Paying
+
+Currently, iff a transaction's fee-payer does not have enough funds to pay the
+fee, the transaction cannot be included in a block. With this proposal, it is
+possible for such a transaction to be included in the block, and there are
+three different edge-cases to consider:
+
+1. The fee-payer account does not exist (0 lamports)
+2. The fee-payer account does not have enough funds for the entire fee
+3. The fee-payer account has enough funds for the fee, but would no longer be
+rent-exempt
+
+In case 1, the transaction should simply be ignored for execution, and have no
```

Comment:
Should also consider that base fee will eventually change and could be much greater than now.